vyper/builtin_functions/signatures.py

```diff
@@ -71,6 +76,64 @@ def decorator_fn(self, node, context):
             if k not in kwsubs:
                 kwsubs[k] = expected_arg.default
 
+        for k, v in kwsubs.items():
+            if isinstance(v, IRnode):
+                v.annotation = k
+
         return wrapped_fn(self, node, subs, kwsubs, context)
 
     return decorator_fn
+
+
+# TODO: rename me to BuiltinFunction
+class _SimpleBuiltinFunction:
+
+    _has_varargs = False
+    _kwargs: Dict[str, KwargSettings] = {}
+
+    def _validate_arg_types(self, node):
+        num_args = len(self._inputs)  # the number of args the signature indicates
+
+        expect_num_args = num_args
+        if self._has_varargs:
+            # note special meaning for -1 in validate_call_args API
+            expect_num_args = (num_args, -1)
+
+        validate_call_args(node, expect_num_args, getattr(self, "_kwargs", []))
+
+        for arg, (_, expected) in zip(node.args, self._inputs):
+            validate_expected_type(arg, expected)
+
+        # typecheck varargs. we don't have type info from the signature,
+        # so ensure that the types of the args can be inferred exactly.
+        varargs = node.args[num_args:]
+        if len(varargs) > 0:
```

Comment:
Maybe check `if len(node.args) > len(self._inputs):`  directly

Reply:
i used `varargs` because it's reused two lines down. but i forgot to reuse it! fixing